### ISSUE TITLE
fix: show error message when API calls fail

### ISSUE THREAD

**Issue by nicholaschiang** [tutorbookapp/tutorbook#28]
Right now, we're just catching and logging error messages but we aren't actually showing anything to the client (so they think that their request succeeded if they don't open up a console and look at the logs) in:
- The `PupilForm` in `src/pupil-form/lib/covid-pupil-form.tsx` that calls `/api/user`
- The `TutorForm` in `src/tutor-form/lib/covid-tutor-form.tsx` that calls `/api/user`
- The `HeroForm` in `src/hero-form/lib/hero-form.tsx` that calls `/api/user`
- The `SearchResults` in `src/search/lib/results.tsx` that calls `/api/search`
- The `UserDialog` in `src/user-dialog/lib/user-dialog.tsx` that calls `/api/appt`

We need to show some type of pop-up dialog explaining what went wrong with each error code and how they can address it (such a dialog should also include a CTA button to open our [Intercom](https://intercom.com) messenger and chat with support).

**Comment by nicholaschiang** [comment]
Comment from #52 which was a duplicate of this issue:

> **The problem**
> Right now, if an `/api/` request fails, we still show the checkmark and the client thinks that the request succeeded (unless, for some strange reason, they think to open up the console and look at the error logs).
> 
> **The solution**
Actually implement the `/api/` endpoint requirements that are [outlined here](https://github.com/tutorbookapp/covid-tutoring/tree/develop/src/pages/api#readme):
> > Each of these API functions catches all known errors and sends them back to the client in plain English (i.e. you can display res.data to the user when you receive a 400 or 500 HTTP error code).
> 
> Then, just display the error codes that are sent back by our API endpoints. 
> 
> Note that you should consult with design on how to best show those error codes (as that's very much a front-end job).

**Comment by nicholaschiang** [comment]
This was also (partially) addressed in [the new redesign](https://tutorbook.org) (released in `v0.7.0`). Our error messages in the new `UserDialog` look like:

![image](https://user-images.githubusercontent.com/20798889/83459625-9a6a0b00-a419-11ea-9afd-18dc158c3a52.png)

Note that we still need to show errors that occur in the `VolunteerForm` when a mentor/tutor is signing up.

**Comment by nicholaschiang** [comment]
Closing this because I'm pretty sure we've addressed most of the scenarios where this is necessary:
- The `RequestDialog` now shows an error message below the "Signup and request" button (see last comment).
- The `VerificationDialog` shows an error message below the verifications grid (also like above).
- The `SignupPage` shows a similar error message below the "Continue with Google" button.

We've yet to merge #70, but I don't think that'll be necessary at this point.
